src/Client.php

```diff
@@ -84,16 +85,16 @@ public function setQrcodeCallback($closure)
 
     public function sessionStart()
     {
-        if (!$this->loadSessionFronFile()) {
+        if (!$this->loadSessionFromFile()) {
             $this->login();
         }
         $this->WsapiWrapper = new Wrapper($this->client);
     }
 
     public function loop($callback)
     {
-        $this->running = true;
-        while ($this->running) {
+        $running = true;
+        while ($running) {
             $jsMessages = $this->WsapiWrapper->getBufferedNewMessages();
             if ($jsMessages) {
                 $callback($this, $jsMessages);
```

Comment:
The client instance is passed to the callback, if the callback marks "stop" on the client, the loop should be interrupted

Reply:
Looking at this I have no idea why I changed that.  I'll put it back shortly.  Apologies.

Reply:
I've now reverted that change, apologies again.